jaeger-core/src/main/java/com/uber/jaeger/Configuration.java

```diff
@@ -165,6 +188,26 @@ public static Configuration fromEnv() {
   }
 
   public synchronized io.opentracing.Tracer getTracer() {
+    if (disableGlobalTracer) {
+      return getLocalTracer();
+    }
+
+    return getGlobalTracer();
+  }
+
+  private io.opentracing.Tracer getGlobalTracer() {
+    if (GlobalTracer.isRegistered()) {
+      return GlobalTracer.get();
+    }
+
+    Tracer tracer = getTracerBuilder().build();
+    GlobalTracer.register(tracer);
+    log.info("Initialized and registered global tracer={}", tracer);
+    return tracer;
+  }
```

Comment:
this seems unnecessary complicated - why not keep the single getTracer method and in the lazy init path (otherwise unchanged) just add `if (!disableGlobalTracer) { GlobalTracer.register(tracer);}`? Everything else keeps working off the local `tracer` var.

Reply:
I wrote it this way because I wanted to remove tracer from jaeger config in a future release, and instead depend on opentracing's GlobalTracer for management of the singleton. 

Do you think we need to continue support holding on to a tracer in the config?
 
The only case where I see this being useful is for advanced users who need multiple tracers in the same application. I expect that such users to use `getTracerBuilder` and not maintain multiple copies of JaegerConfiguration.

Reply:
if or when we decide on that separation (I am not convinced of its value), we can do that, but right now I don't see a value in this big change that can be instead achieved with just 3 lines.

Reply:
ok